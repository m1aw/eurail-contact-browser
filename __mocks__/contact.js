export const contactMock = {
    "gender": "female",
    "name": {
        "title": "Ms",
        "first": "Loane",
        "last": "Mercier"
    },
    "location": {
        "street": {
            "number": 1826,
            "name": "Boulevard de Balmont"
        },
        "city": "NÃ®mes",
        "state": "Cher",
        "country": "France",
        "postcode": 92735,
        "coordinates": {
            "latitude": "2.0946",
            "longitude": "35.9945"
        },
        "timezone": {
            "offset": "+9:30",
            "description": "Adelaide, Darwin"
        }
    },
    "email": "Loane.Mercier@example.com",
    "login": {
        "uuid": "9ace22a2-eb1c-4a3a-99f1-d4949847ef13",
        "username": "beautifulsnake364",
        "password": "virginie",
        "salt": "SdXNnOtL",
        "md5": "aec794b5992ec92967eca0fbaf945be5",
        "sha1": "feaef2a39c6e9871a3f9a33591411cddd017a108",
        "sha256": "76edff81182948ea707b98066303bf841579796049310043f3119903c4b5da7a"
    },
    "dob": {
        "date": "1975-05-17T00:40:51.032Z",
        "age": 44
    },
    "registered": {
        "date": "2006-08-20T21:17:39.206Z",
        "age": 13
    },
    "phone": "03-24-99-12-06",
    "cell": "06-99-28-89-03",
    "id": {
        "name": "INSEE",
        "value": "2NNaN99001462 16"
    },
    "picture": {
        "large": "https://randomuser.me/api/portraits/women/58.jpg",
        "medium": "https://randomuser.me/api/portraits/med/women/58.jpg",
        "thumbnail": "https://randomuser.me/api/portraits/thumb/women/58.jpg"
    },
    "nat": "FR"
}
export const contactMock2 =  {
    "gender": "male",
    "name": {
        "title": "Mr",
        "first": "Arnold",
        "last": "Edwards"
    },
    "location": {
        "street": {
            "number": 2415,
            "name": "Forest Ln"
        },
        "city": "Sioux Falls",
        "state": "Arizona",
        "country": "United States",
        "postcode": 46902,
        "coordinates": {
            "latitude": "88.1574",
            "longitude": "-111.5056"
        },
        "timezone": {
            "offset": "+9:00",
            "description": "Tokyo, Seoul, Osaka, Sapporo, Yakutsk"
        }
    },
    "email": "Arnold.Edwards@example.com",
    "login": {
        "uuid": "13e34d06-3c8c-4ada-8423-d74e4b81a569",
        "username": "blackduck879",
        "password": "dawg",
        "salt": "oXFVQ2NF",
        "md5": "0958ae902fdbd362f0e4776007e7eeda",
        "sha1": "6243a80c7f11e207c51d9715a178310e95f1a68c",
        "sha256": "82453bdb4b0f2b3390a0973a986b0281f857926312e3e1a2a4bfba36e2eda388"
    },
    "dob": {
        "date": "1970-05-26T14:26:27.822Z",
        "age": 49
    },
    "registered": {
        "date": "2007-04-02T22:25:47.631Z",
        "age": 12
    },
    "phone": "(365)-163-3583",
    "cell": "(476)-642-4385",
    "id": {
        "name": "SSN",
        "value": "907-51-4901"
    },
    "picture": {
        "large": "https://randomuser.me/api/portraits/men/89.jpg",
        "medium": "https://randomuser.me/api/portraits/med/men/89.jpg",
        "thumbnail": "https://randomuser.me/api/portraits/thumb/men/89.jpg"
    },
    "nat": "US"
}