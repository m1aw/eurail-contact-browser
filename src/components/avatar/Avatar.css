.Avatar img {
    border-radius: 50%;
}