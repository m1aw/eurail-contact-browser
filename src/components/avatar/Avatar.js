import React from 'react'
import './Avatar.css'

export const Avatar = ({contact}) => {
    return (
        <div className={'Avatar'}>
            <img alt={'contact avatar'} src={contact.picture.medium}/>
        </div>
    )
}
