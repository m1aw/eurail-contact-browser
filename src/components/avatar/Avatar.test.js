import React from 'react';
import renderer from 'react-test-renderer';
import {Avatar} from "./Avatar";
import {contactMock} from '../../../__mocks__/contact'

test('Avatar should match snapshot', () => {
    const component = renderer.create(
        <Avatar contact={contactMock} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
