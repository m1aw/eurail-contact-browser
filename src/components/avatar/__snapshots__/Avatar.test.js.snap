// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match snapshot 1`] = `
<div
  className="Avatar"
>
  <img
    alt="contact avatar"
    src="https://randomuser.me/api/portraits/med/women/58.jpg"
  />
</div>
`;
